homeassistant/components/tplink/__init__.py

```diff
@@ -115,6 +187,9 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
         hass_data.pop(entry.entry_id)
     await device.protocol.close()
+
+    if len(hass.config_entries.async_entries(DOMAIN)) == 1:
+        await _get_store(hass).async_remove()
     return unload_ok
```

Comment:
This should go in `async_remove_entry`

See example in `homekit_controller`

Reply:
Done